service/history/replication/task_processor_test.go

```diff
@@ -601,3 +602,108 @@ func (s *taskProcessorSuite) TestShouldRetryDLQ() {
 	s.False(s.taskProcessor.shouldRetryDLQ(&types.ServiceBusyError{}))
 	s.False(s.taskProcessor.shouldRetryDLQ(fmt.Errorf("error after done channel closed should NOT be retried")))
 }
+
+func TestProcessorLoop_TaskExecuteFailed_ShardChangeErr(t *testing.T) {
+
+	for i := 0; i < 10; i++ {
```

Comment:
I don't think we need to loop 10 times here. It increases overall test duration. Single run should give sufficient coverage - let's adjust mock call times accordingly and avoid AnyTimes